roles/servicetelemetry/templates/manifest_smartgateway_events.j2

```diff
@@ -1,27 +1,28 @@
 apiVersion: smartgateway.infra.watch/v2
 kind: SmartGateway
 metadata:
-  name: '{{ this_smartgateway }}'
-  namespace: '{{ ansible_operator_meta.namespace }}'
+  name: {{ this_smartgateway }}
+  namespace: {{ ansible_operator_meta.namespace }}
 spec:
 {% if this_collector.debug_enabled is defined and this_collector.debug_enabled %}
-  logLevel: "debug"
+  logLevel: debug
 {% else %}
-  logLevel: "info"
+  logLevel: info
 {% endif %}
   handleErrors: true
   size: {{ smartgateway_deployment_size }}
   applications:
   - config: |
-      hostURL: https://elasticsearch-es-http.{{ ansible_operator_meta.namespace }}.svc.cluster.local:9200
-      useTLS: true
-      tlsClientCert: /config/certs/tls.crt
-      tlsClientKey: /config/certs/tls.key
+      hostURL: {{ elasticsearch.ext_host_url }}
+      useTLS: {{ elasticsearch.ext_use_tls }}
+      tlsClientCert: /config/certs/user.crt
+      tlsClientKey: /config/certs/user.key
       tlsCaCert: /config/certs/ca.crt
-      tlsServerName: 'elasticsearch-es-http.{{ ansible_operator_meta.namespace }}.svc.cluster.local'
-      user: '{{ elastic_user | default('elastic') }}'
-      password: '{{ elastic_pass | default('') }}'
-      useBasicAuth: true
+      tlsServerName: {{ (elasticsearch.ext_tls_server_name != "") | ternary(elasticsearch.ext_tls_server_name, elasticsearch.ext_host_url.rpartition("//")[-1].partition(":")[0]) }}
```

Comment:
I wasn't sure if `rpartition` was a typo, but it does not appear to be. See https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_variables.html#referencing-key-value-dictionary-variables

I also saw this note which was interesting:

> Use bracket notation if you use keys which start and end with two underscores (which are reserved for special meanings in python) or are any of the known public attributes:

It might be worth a note here just to document exactly what is being parsed here for future reference. It's complex enough that my brain hurts (although it's almost lunch).

Reply:
Hmm, I can't add a comment inside a template, but could move this logic into a pair of tasks to make them clearer and comment there? It probably won't be today, but I can come back and add that.

To help others, this is the logic spelled out:

- If ext_tls_server_name is not empty, use it
- Else, use the ext_host_url, returning the substring to the right of '//' and the left of ':'